obspy/io/ah/core.py

```diff
@@ -257,7 +257,10 @@ def _pack_trace_with_ah_dict(trace, packer):
         packer.pack_int(CODESIZE)
         packer.pack_fstring(CODESIZE, tr.stats.ah.station.code)
         packer.pack_int(CHANSIZE)
-        packer.pack_fstring(CHANSIZE, tr.stats.ah.station.channel)
+        try:
+            packer.pack_fstring(CHANSIZE, tr.stats.channel)
+        except:
+            packer.pack_fstring(CHANSIZE, tr.stats.ah.station.channel)
```

Comment:
@s-schneider what is the logic behind this? Packing/unpacking should always go via `Trace.stats.station/channel/...` so I do not see the need to lookup those codes from within `tr.stats.ah`.